pkg/engine/planner/physical/optimizer.go

```diff
@@ -198,26 +199,54 @@ func (r *projectionPushdown) apply(node Node) bool {
 		// Always project timestamp column
 		projections[len(node.PartitionBy)] = &ColumnExpr{Ref: types.ColumnRef{Column: types.ColumnNameBuiltinTimestamp, Type: types.ColumnTypeBuiltin}}
 
-		return r.applyProjectionPushdown(node, projections, false)
+		ambiguousProjections := collectAmbiguousColumns(node.PartitionBy)
+		return r.applyProjectionPushdown(node, projections, ambiguousProjections, false)
+	case *VectorAggregation:
+		if len(node.GroupBy) == 0 || node.Operation != types.VectorAggregationTypeSum {
+			return false
+		}
+
+		projections := make([]ColumnExpression, len(node.GroupBy)+1)
+		copy(projections, node.GroupBy)
+		// Always project timestamp column
+		projections[len(node.GroupBy)] = &ColumnExpr{Ref: types.ColumnRef{Column: types.ColumnNameBuiltinTimestamp, Type: types.ColumnTypeBuiltin}}
+		ambiguousProjections := collectAmbiguousColumns(node.GroupBy)
+
+		return r.applyProjectionPushdown(node, projections, ambiguousProjections, false)
+
+	case *ParseNode:
+		// Parse nodes need the message column to extract fields from log lines
+		projections := []ColumnExpression{
+			&ColumnExpr{Ref: types.ColumnRef{Column: types.ColumnNameBuiltinMessage, Type: types.ColumnTypeBuiltin}},
+		}
+		// Apply projection pushdown, but only add to existing projections (for metric queries)
+		return r.applyProjectionPushdown(node, projections, nil, true)
 	case *Filter:
 		projections := extractColumnsFromPredicates(node.Predicates)
 		if len(projections) == 0 {
 			return false
 		}
+		ambiguousProjections := collectAmbiguousColumns(extractColumnsFromPredicates(node.Predicates))
 
 		// Filter nodes should only add their predicate columns to projections when
 		// there's already a projection list in the plan (indicating a metric query).
 		// For log queries that read all columns, filter columns should not be projected.
 		//
 		// Setting applyIfNotEmpty argument as true for this reason.
-		return r.applyProjectionPushdown(node, projections, true)
+		return r.applyProjectionPushdown(node, projections, ambiguousProjections, true)
 	}
 	return false
 }
 
 // applyProjectionPushdown applies the projection pushdown rule to the given node.
+// we can't push all projections down to the scan node, since some may be referencing parsed columns.
 // if applyIfNotEmpty is true, it will apply the projection pushdown only if the node has existing projections.
-func (r *projectionPushdown) applyProjectionPushdown(node Node, projections []ColumnExpression, applyIfNotEmpty bool) bool {
+func (r *projectionPushdown) applyProjectionPushdown(
+	node Node,
+	projections []ColumnExpression,
+	ambiguousProjections []ColumnExpression,
```

Comment:
nit: rather than passing ambiguousProjections as an arugment (which is always a subset of projections but not used for DataObjScan), I think the `ParseNode` case should build the subset itself to keep the number of parameters minimal.